model/execution/implementation/Lti1p3ContextCacheRepository.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2023  (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\ltiDeliveryProvider\model\execution\implementation;
+
+use oat\ltiDeliveryProvider\model\execution\LtiContextRepositoryInterface;
+use oat\taoDelivery\model\execution\DeliveryExecutionInterface;
+use oat\taoLti\models\classes\LtiLaunchData;
+use Psr\Cache\CacheItemPoolInterface;
+
+class Lti1p3ContextCacheRepository implements LtiContextRepositoryInterface
+{
+    private CacheItemPoolInterface $cache;
+
+    private const KEY_PREFIX = 'de_lti1p3context_';
+
+    public function __construct(?CacheItemPoolInterface $cache = null)
+    {
+        $this->cache = $cache;
+    }
+
+    public function findByDeliveryExecution(DeliveryExecutionInterface $deliveryExecution): ?LtiLaunchData
+    {
+
+        $item = $this->cache->getItem($this->buildKey($deliveryExecution->getIdentifier()));
+
+        if ($data = $item->get()) {
```

Comment:
does $this->cache->getItem() always return objects? If not and you get null this code will fail. You can add if ($item && $item->get()) for quick check.

Reply:
_does $this->cache->getItem() always return objects?_ 
According to the interface it should, unless someone create a wrong implementation. Good place to catch if it happen :)